team/challenges/systems/worker.tex

```diff
@@ -0,0 +1,268 @@
+    \item Not communicating. Candidates who submitted all the code to master branch, which does not give us the ability to provide feedback on the various implementation phases. Because we are a distributed team, structured communication is critical to us.
+    \item Implementing custom security algorithms/authentication schemes is always a bad idea unless you are a trained security researcher/engineer. It is definitely a bad idea for this task - try to stick to industry proven security methods as much as possible.
+    \end{itemize}
+
+\subsection{Scoring}
+    
+We want to be as transparent as possible on how we will be scoring your submission.
+The following table provides a description of different areas you will be evaluated on and how they will affect your overall score.
+
+\begin{center}
+\rowcolors{2}{gray!80!gray!50}{gray!50!gray!20}
+\begin{tabular}{ | m{25em} | m{5em}| m{5em} | }
+  \hline
+  \rowcolor{blue!60!black!10}
+  Description & Points Awarded & Points Subtracted \\
+  \hline
+  The submitted code has a clear and modular structure & +1 & -1 \\
+  \hline   
+  The candidate communicated their progress during the interview & +1 & -1 \\
+  \hline
+  The program builds are reproducible & +1 & -1 \\
+  \hline
+  README provides clear instructions & +1 & -1 \\
+  \hline
+  The candidate outlined the key design points in the design document & +1 & -1 \\
+  \hline
+  The code has no obvious data races and deadlocks & +1 & -1 \\
+  \hline
+  The code provides examples of tests covering key components & +1 & -1 \\
+  \hline
+  The code provides clear error handling and reporting & +1 & -1 \\
+  \hline
+  The program is working according to the specification & +1 & -1 \\
+  \hline
+  The candidate demonstrates ability to handle and apply feedback & +1 & -1 \\
+  \hline
+  The client-server communication is implemented in a secure way & +1 & -1 \\
+  \hline
+\end{tabular}
+\end{center}
+
+\subsection{Scoring}
+
+It is OK to ask the interview team questions. Some folks stay away from
+asking questions to avoid appearing less experienced, so we provide examples of questions
+to ask and questions we expect candidates to figure out on their own.
+
+Here is a great question to ask:
+
+``Is it OK to pre-generate secret data and put the secrets in the repository for the purposes of POC? I will add a note that we will auto-generate secrets in the future.''
+
+It demonstrates that you thought about this problem domain, recognize the trade off and save you and the team time by not implementing it.
+
+This is the question we expect candidates to figure out on their own:
+
+``What version of Go should I use? What dependency manager should I use?''
+
+Unless specified in the requirements, pick the solution that works best for you.
+
+\section{Tools}
+
+This task should be implemented in Go, C++ or Rust and should work on 64-bit Linux machines
+
+\section{Timing}
+
+It should take you from 4 to 24 full hours to complete the challenge.
```

Comment:
I really like that we're expressing this here and giving a realistic expectation of how long we expect someone to spend on the challenge. Having read a lot HN/Reddit comments, however, a lot of engineers would baulk at the thought of spending this much time on something that might not ultimately get them a job. I'm not sure what we can do about this, but wanted to raise it.